File: src/main/java/visitor/Printer.java
package visitor;

import calculator.MyNumber;
import calculator.Notation;
import calculator.Operation;

public class Printer extends Visitor{

    public final Notation notation;

    public Printer(Notation notation) {
        this.notation = notation;
    }

    @Override
    public void visit(MyNumber n) {
        // Do nothing
    }

    @Override
    public void visit(Operation o) {
        o.setNotation(notation);
    }
}

File: src/main/python/visitor/printer.py
from src.main.python.visitor.visitor import Visitor


class Printer(Visitor):
    def __init__(self, notation):
        self.result = ""
        self.__notation = notation

    def visit_operation(self, op):
        op.set_notation(self.__notation)

    def visit_my_number(self, number):
        pass

    def visit_linear_solution(self, solution):
        self.result += str(solution)

    def visit_matrix(self, matrix):
        """The Visitor can traverse a matrix"""
        pass
